### PR TITLE
fixed saving to card on console

### PR DESCRIPTION
**Pull request by kipcode66**
Saving GZ's settings on card now works for console. But I couldn't get it to work by dynamically allocating a buffer based on the sector size. So for now, it allocates a buffer statically on the stack.

### CODE REVIEW

**Comment by kipcode66** [comment]
I got some advice on how to improve it further, please wait for my next commit before deciding to merge or not.

**Review by Pheenoh — APPROVED** [review]

**Pull request merged**
